compile.sh

```diff
@@ -46,46 +63,127 @@ function require_programs() {
   fi
 }
 
+############################################################
+# Generate CPD report based on language in the report path.
+# For Scala, also remove duplicates generated due to license
+# header as they are false negatives.
+#
+# Arguments:
+#   arg1: Language (one of Java or Scala)
+# Returns:
+#   None
+############################################################
+function generateCPDReportByLanguage() {
+  cpd_result_file=$(getCPDReportName $CPD_REPORT_BASE_PATH $1)
+  mv $CPD_REPORT_PATH $cpd_result_file
+  if [ $1 = "Scala" ]; then
+    removeLicenseHeaderDuplicates $cpd_result_file
+  fi
+  echo "CPD report generated at path $cpd_result_file"
```

Comment:
checkCPDReport should be called here and it should fail in local as well. Otherwise it would be difficult to fix.

Reply:
Ok. Will fail the build in compile.sh based on baseline/threshold used in travis CI build by moving the constants to another script/configuration